src/Database/Firebase.hs

```diff
@@ -32,22 +32,75 @@ query = Query Nothing Nothing Nothing Nothing
 key :: ToJSON a => a -> Key
 key = Key
 
-get :: (MonadIO m, MonadError e m, MonadReader r m, AsHttpError e, HasFirebase r, HasHttpCfg r, FromJSON a) => Location -> Maybe Query -> m a
+get :: (
+  MonadIO m, 
+  MonadError e m, 
+  MonadReader r m,
+  AsHttpError e,
+  HasFirebase r,
+  HasHttpCfg r,
+  FromJSON a) => 
+    Location -> 
```

Comment:
Oh jeez, right. Now I understand what this is trying to do. Sorry, I thought it was trying to be this: https://hackage.haskell.org/package/path-pieces , but now I get that it's trying to be an interface for an entire arbitrary type into a location path.